packages/start/README.md

```diff
@@ -66,6 +66,117 @@ const [theme, setTheme] = createUserTheme("cookieName", {
 theme(); // => "light" | "dark"
 ```
 
+
+## Examples
+
+### `root.tsx`
+
+This is the main entry file for SolidStart demonstrating usage of the `createUserTheme` function.
+- Initializes the theme with the `createUserTheme` function.
+- Toggles between the "dark" and "light" themes using a button.
+- Sets the `data-theme` attribute adding compatibility between `createUserTheme` and libraries like [DaisyUI](https://daisyui.com/).
+- Includes a clean implementation of theming through a `Body` class.
+- Provides additional examples on using variant classes to style components.
+
+```tsx
+import { Suspense } from "solid-js";
+import {
+  A,
+  Body,
+  ErrorBoundary,
+  FileRoutes,
+  Head,
+  Html,
+  Meta,
+  Routes,
+  Scripts,
+  Title,
+} from "solid-start";
+import Counter from "./components/Counter";
+import "./root.css";
+import { createUserTheme } from '~/primitives/start';
+export default function Root() {
+  const [theme, setTheme] = createUserTheme("dark-light-theme", "dark");
+  const toggleTheme = () => setTheme(currentTheme => currentTheme === "dark" ? "light" : "dark");
+  return (
+    // Set data-theme when using libraries like Daisyui:
+    <Html lang="en" data-theme={theme()}>
+      <Head>
+        <Title>SolidStart - Bare</Title>
+        <Meta charset="utf-8" />
+        <Meta name="viewport" content="width=device-width, initial-scale=1" />
+      </Head>
+      // Apply theme as a class to the Body element:
+      <Body class={theme()}>
```

Comment:
Considering the demonstration of `classList` at the component level I decided to follow your recommendation and keep this simple.